io/plugins/eu.esdihumboldt.hale.io.jdbc.mssql/src/eu/esdihumboldt/hale/io/jdbc/mssql/MsSqlGeometries.java

```diff
@@ -0,0 +1,231 @@
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.io.WKTReader;
+
+import de.fhg.igd.slf4jplus.ALogger;
+import de.fhg.igd.slf4jplus.ALoggerFactory;
+import eu.esdihumboldt.hale.common.instance.geometry.DefaultGeometryProperty;
+import eu.esdihumboldt.hale.common.instance.geometry.impl.CodeDefinition;
+import eu.esdihumboldt.hale.common.instance.geometry.impl.WKTDefinition;
+import eu.esdihumboldt.hale.common.schema.geometry.CRSDefinition;
+import eu.esdihumboldt.hale.common.schema.geometry.GeometryProperty;
+import eu.esdihumboldt.hale.common.schema.model.TypeDefinition;
+import eu.esdihumboldt.hale.common.schema.model.impl.DefaultTypeDefinition;
+import eu.esdihumboldt.hale.io.jdbc.GeometryAdvisor;
+import eu.esdihumboldt.hale.io.jdbc.constraints.GeometryMetadata;
+import eu.esdihumboldt.hale.io.jdbc.mssql.util.SRSUtil;
+import schemacrawler.schema.Column;
+import schemacrawler.schema.ColumnDataType;
+
+/**
+ * Geometry advisor for MSSQL database
+ * 
+ * @author Arun
+ */
+public class MsSqlGeometries implements GeometryAdvisor<SQLServerConnection> {
+
+	private static final ALogger log = ALoggerFactory.getLogger(MsSqlGeometries.class);
+
+	@Override
+	public Object convertGeometry(GeometryProperty<?> geom, TypeDefinition columnType,
+			SQLServerConnection connection) throws Exception {
+		if (geom instanceof DefaultGeometryProperty) {
+
+			// We need Column Data type
+			String columnDataType = columnType.getName().getLocalPart();
+
+			CoordinateReferenceSystem targetCRS = null;
+
+			Statement stmt = null;
+			ResultSet rs = null;
+			try {
+				int srId = 4326;
+
+				if (geom.getCRSDefinition() != null) {
+
+					// SRS Code
+					String srsName = CRS.toSRS(geom.getCRSDefinition().getCRS());
+
+					// Todo:: Do we really need transformation for 'geometry'
+					// columnDataType instances? because this type represents
+					// data in a Euclidean (flat) coordinate system.
+
+					// getting Axis order of geometry CRS
+					CRS.AxisOrder axisOrder = CRS.getAxisOrder(geom.getCRSDefinition().getCRS());
+					// if axis order is not x/y ordering then will need to
+					// transform geometry to target crs with longiture first
+					if (axisOrder != CRS.AxisOrder.EAST_NORTH) {
+						if (srsName != null && srsName.startsWith("EPSG")) {
+							targetCRS = CRS.decode(srsName, true);
+						}
+					}
+
+					try {
+						if (srsName != null) {
+							final int index = srsName.lastIndexOf(':');
+							if (index > 0) {
+								srsName = srsName.substring(index + 1).trim();
+							}
+							srId = Integer.parseInt(srsName);
+						}
+					} catch (NumberFormatException nfEx) {
+						// TODO::Using UI ask user to enter default SRId, if
+						// can not extract it.
+					}
+				}
+
+				Geometry targetGeometry;
+				if (targetCRS != null) {
+					MathTransform transform = CRS
+							.findMathTransform(geom.getCRSDefinition().getCRS(), targetCRS);
```

Comment:
findMathTransform can be very costly if done many times. We should look into a global mechanism that we can use where there is an automatic caching done.
We would need that as well for the reprojection Groovy helper function that is to be implemented in context of another issue.